### PR TITLE
Add Support For `...` Passthrgouh

### PR DESCRIPTION
**Pull request by ytaben**
Currently, MethodParameters has a `pass_through?` method which supports `def foo(*)` cases.

Ruby 2.7, however, adds a new method of passthrough, namely `def foo(...)`.

MethodParameters is not aware of that which results in kwargs auto inject mistakenly believing that the parent class supports `splat` arguments.

I didn't report this as an issue since I already had a fix but let me know if you need me to do so or if tests are required in more places.

Edit: Looks like this fix _only_ works on 2.7. I'll look into other ruby versions.

### CODE REVIEW

File: lib/dry/auto_inject/method_parameters.rb [hunk 1]
```diff
@@ -87,7 +87,7 @@ def length
       end
 
       def pass_through?
-        parameters.eql?(PASS_THROUGH)
+        PASS_THROUGH.any? { |pass_through_params| parameters.eql?(pass_through_params) }
```
Comment (solnic) [inline lib/dry/auto_inject/method_parameters.rb:90]:
```suggestion
        PASS_THROUGH.include?(parameters)
```

**Review by solnic — APPROVED** [review]
Approved with a small suggestion. Thank you ❤️

**Comment by solnic** [comment]
> Edit: Looks like this fix only works on 2.7. I'll look into other ruby versions.

Oops. I didn't notice this hah. OK so this has to be obviously addressed

**Comment by solnic** [comment]
@ytaben thanks! We can deal with this here I think. It can be broken down into two PRs later if it turns out to be too big.

**Comment by ytaben** [comment]
@solnic 

I've looked more into this issue and I don't see how we can easily fix it in a non complicated way.

At the moment we define the initializer, we don't know what the user-defined initializer will be.
There is a corner case where kwargs auto inject won't work when users define an initializer _without_ kwargs _after_ the injector module is included.

Realistically, I don't see a way to reliably handle every corner case like this for every ruby version as I'm sure if I poke a bit more - I can make even more specs that don't work.

I'd rely on clients ensuring that their code works via their own tests suites (this kind of stuff should be code with the most basic tests).

Meanwhile, I modified the initializer in the spec to accept kwargs which works in every ruby version I tried.

Let me know what you think.

**Comment by solnic** [comment]
@ytaben I think we will never support all possible cases so this is fine. Personally, I'm quite convinced that eventually this stuff will just use dry-effects anyway and this PR is another reason why we should do it.

File: spec/integration/kwargs_spec.rb [hunk 2]
```diff
@@ -23,8 +23,8 @@ module Test
 
       attr_reader :block
 
-      def initialize(*args, &block)
-        super(*args)
+      def initialize(*args, **kwargs, &block)
+        super(*args, **kwargs)
```
Comment (solnic) [inline spec/integration/kwargs_spec.rb:27]: Would it make things simpler if this just called `super(**kwargs)`? I think it's absolutely reasonable to expect people to do this. If you a) include auto-inject module b) override constructor then you are on your own. The whole point of using dry-auto-inject is to not have to define constructors. It's allowed to do that of course, but it should be an exception.

Comment (ytaben) [reply spec/integration/kwargs_spec.rb:27]: @solnic 
I don't think it would make any difference. This already works with 2.6, 2.7 and 3.0. Since specs emulate client code - it can be either way.

**Comment by ytaben** [comment]
@solnic 
Do we need anything else for this PR to go in?

**Comment by solnic** [comment]
@ytaben thanks for working on this. I'll push a new release this week.
